### ISSUE TITLE
Rule S109: False positive on enums

### ISSUE THREAD

**Issue by olsh** [SonarSource/sonar-csharp#1356]
### Description

Rule S109 detects false positive on enums.

### Repro steps

1. Create an enum:

```csharp
    public enum CreditRole
    {
        Actor = 1,
        Director = 2,
        Producer = 3,
        Writer = 4
    }
```

2. Run an analysis with Rule S109 enabled

3. SonarQube reports issues  
`Assign this magic number '2' to a well-named (variable|constant), and use the (variable|constant) instead.`

### Expected behavior

The magic numbers are not detected.

### Actual behavior

The magic numbers are detected.

### Related information

* SonarC# 7.0.1
* MSBuild Scanner 4.2

**Comment by Evangelink** [comment]
Hi,

Thanks for the feedback. The fix for this issue will be available in 7.1 (see #1244)

**Comment by ievgennaida** [comment]
Looks like still there 2020
typescript:S109???
Example:

enum HttpStatusCode {
    CONTINUE = 100,
}

**Comment by andrei-epure-sonarsource** [comment]
Hi @ievgennaida , this is the repository for our sonar-dotnet analyzers, for C# and VB .NET.
For typescript, please go on our community forum and report the FP https://community.sonarsource.com/c/bug/fp/7
Thanks,
Andrei
